Add WP_ENV_MULTISITE environment variable.

spacedmonkey opened the pull request:
In https://github.com/WordPress/gutenberg/pull/67845 support for `multisite` was added in option in `wp-env` setups.

Related: https://github.com/WordPress/gutenberg/pull/68125

<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
This PR, it adds the functionality to can be specified via the `WP_ENV_MULTISITE` environment variable or in the configuration file. 

## Why?
It makes it easier to define it the site is multisite in a CI context. 

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

<!-- If you would like to upload screenshots, feel free to use the table below when it is useful to show the difference between before and after the change. -->

|Before|After|
|-|-|
|<!-- Before screenshot here -->|<!-- After screenshot here -->|

spacedmonkey committed:
Add WP_ENV_MULTISITE envoriment variable.

Introduce a new `multisite` option in `wp-env` to enable multisite setups. This can be specified via the `WP_ENV_MULTISITE` environment variable or in the configuration file. Updated documentation to include usage instructions.

spacedmonkey commented on packages/env/README.md:
````diff
@@ -701,6 +701,17 @@ You can tell `wp-env` to use a specific PHP version for compatibility and testin
+
+```json
+{
+	"multsite": true,
````
```suggestion
	"multisite": true,
```

spacedmonkey committed:
Apply suggestions from code review

swissspidy commented on packages/env/lib/config/get-config-from-environment-vars.js:
```diff
@@ -23,6 +23,7 @@ const { checkPort, checkVersion, checkString } = require( './validate-config' );
  * @property {?number}                  phpmyadminPort   An override for the development environment's phpMyAdmin port.
  * @property {?WPSource}                coreSource       An override for all environment's coreSource.
  * @property {?string}                  phpVersion       An override for all environment's PHP version.
+ * @property {?bool}                  multisite       An override for if environmen should be multisite.
```
Incorrect indentation

spacedmonkey replied:
```suggestion
 * @property {?bool}                    multisite       An override for if environmen should be multisite.
```

swissspidy commented on packages/env/README.md:
````diff
@@ -701,6 +701,17 @@ You can tell `wp-env` to use a specific PHP version for compatibility and testin
 }
 ```
 
+### Specific if multisite is enabled
````
```suggestion
### Multisite support
```

spacedmonkey committed:
Apply suggestions from code review

swissspidy commented:
This should probably be added to the changelog file too

spacedmonkey committed:
Update packages/env/README.md

Co-authored-by: Pascal Birchler <pascalb@google.com>

spacedmonkey committed:
Add `WP_ENV_MULTISITE` environment variable to override multisite

This enhancement allows overriding the `multisite` option using the `WP_ENV_MULTISITE` variable. It provides greater flexibility for customizing environment configurations.

spacedmonkey commented:
> This should probably be added to the changelog file too

@swissspidy Fixed.

spacedmonkey committed:
Fix type annotation and format multisite environment logic

Corrected the JSDoc type annotation for the multisite property from `bool` to `boolean` for consistency. Added a minor formatting adjustment to the multisite environment variable processing logic.

youknowriad commented:
We discussed this in the original PR. For me it's not really clear why some config should support an environment variable while others shouldn't. (I know this is already the case).

So my opinion is that, I'd prefer to be consistent. Can we avoid having to handle every config option manually and instead have a naming convention. I know some tools that do that.

spacedmonkey commented:
> So my opinion is that, I'd prefer to be consistent. Can we avoid having to handle every config option manually and instead have a naming convention. I know some tools that do that.

I agree for consisancy all or none should have envoriment variable. 

This PR comes out of necessity, I have a requirement to use wp-env in CI to run E2E tests. I need to run the tests against single and multisite. This envoriemnt varible fixes that.

spacedmonkey committed:
Merge branch 'trunk' into fix/multisite-env

swissspidy commented:
> This PR comes out of necessity, I have a requirement to use wp-env in CI to run E2E tests. I need to run the tests against single and multisite. This environment variable fixes that.

`.wp-env.override.json` also works for that, but it's less ergonomic.
